uk.co.compucorp.civicrm.hrleaveandabsences/hrleaveandabsences.php

```diff
@@ -483,6 +483,19 @@ function hrleaveandabsences_addContactMenuActions(ActionsMenu $menu){
   $leaveActionGroup->setWeight(1);
   $menu->addToMainPanel($leaveActionGroup);
 }
+
+/**
+ * Implementation of hook_civicrm_pageRun
+ */
+function hrleaveandabsences_civicrm_pageRun(&$page) {
+  // Adds the baseURL var to all L&A pages
+  CRM_Core_Resources::singleton()->addVars('leaveAndAbsences', [
+    'attachmentToken' => CRM_Core_Page_AJAX_Attachment::createToken(),
+    'baseURL' => CRM_Core_Resources::singleton()->getUrl('uk.co.compucorp.civicrm.hrleaveandabsences'),
+    'contactId' => CRM_Utils_Request::retrieve('cid', 'Integer'),
+    'loggedInUserId' => CRM_Core_Session::getLoggedInContactID(),
+  ]);
+}
```

Comment:
This will include these variables in ALL civicrm pages. Is this really what you want?

Reply:
This is what I'm not sure of. There is a positive side effect though: we wouldn't need to manually add the `leaveAndAbsences` var to specific pages when they are implementing a module from L&A.

Something else that we could do is have a list of pages that the var is allowed to exist, like this:

```php
$allowedPages = [
 ClassName_of_the_Page1,
 ClassName_of_the_Page2,
];

if (!in_array($page, $allowedPages)) {
  return;
}

// ...
```

Reply:
If there's no need to use any of these vars outside L&A pages, then I'd prefer to have them available only in the pages they will be used. If the requirements change in the future, then we can also change the solution.

As for the hook implementation, here's a suggestion. We've been trying to move logic out of the hooks into separate classes. So it would be nice if you could do that here. Check this example: https://github.com/compucorp/civihr/blob/staging/uk.co.compucorp.civicrm.hrcore/hrcore.php#L287-L293